src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkConnector.java

```diff
@@ -48,9 +61,62 @@ public void stop() {
     public ConfigDef config() {
         return CosmosDBSinkConfig.getConfig();
     }
-    
+
     @Override
     public String version() {
         return this.getClass().getPackage().getImplementationVersion();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Config config = super.validate(connectorConfigs);
+        if (config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+            return config;
+        }
+
+        Map<String, ConfigValue> configValues = config.configValues().stream().collect(
+            Collectors.toMap(ConfigValue::name, Function.identity()));
+
+        validateConnection(connectorConfigs, configValues);
+        validateTopicMap(connectorConfigs, configValues);
+
+        return config;
+    }
+
+    private void validateConnection(Map<String, String> connectorConfigs, Map<String, ConfigValue> configValues) {
+        String endpoint = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONN_ENDPOINT_CONF);
+        String key = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONN_KEY_CONF);
+        try {
+            createClient(endpoint, key);
+        } catch (Exception e) {
+            configValues.get(CosmosDBSinkConfig.COSMOS_CONN_ENDPOINT_CONF)
+                .addErrorMessage("Could not connect to endpoint with error: " + e.getMessage());
+            configValues.get(CosmosDBSinkConfig.COSMOS_CONN_KEY_CONF)
+                .addErrorMessage("Could not connect to endpoint with error: " + e.getMessage());
+        }
+    }
+
+    private void validateTopicMap(Map<String, String> connectorConfigs,
+        Map<String, ConfigValue> configValues) {
+
+        String topicMap = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONTAINER_TOPIC_MAP_CONF);
+        try {
+            TopicContainerMap.deserialize(topicMap);
```

Comment:
@ncliang Would it be useful to validate that all the topic included in the `topics` config are specified here? Also, if there is a way to validate if all the specified tables exist, that might be useful as well.
More to add - Can we check write permissions on these tables?

Reply:
@mageshn I think the topics validation might be useful to catch typos in the configuration. I can add that in a followup PR.

For the tables/containers specified in the topic map, I am wondering if it makes more sense to just create them if they do not already exist via a call to `createContainerIfNotExists`. This is done in the source task but not sink. In sink, there is a call to `createDatabaseIfNotExists` to ensure we have a database, but no such call on the container. @ryancrawcour , could you comment on this and whether there might be permission considerations for doing this?

Reply:
@mageshn  tables don't exist as a concept in Cosmos DB. it's like MongoDB in that respect. it has a container for json documents. that container can contain different documents. it is schemalass. 

with the masterkey that you are using to make the connection, you will always have write permission to the container.

Reply:
@ncliang & @mageshn yeah there are definitely more validations we can add. i have left an issue open to track these additional validations. things like database name and the container name etc. can be validated. topic name too. we can do a regex validation against these settings, but we can also attempt a connection to Cosmos and Kafka and ensure the actual entity exists 

so, let's track the additional validations we want to perform (and any recommendations) in #372